Framework/Algorithms/src/ExtractSpectra.cpp

```diff
@@ -156,6 +172,80 @@ void ExtractSpectra::execHistogram() {
   }
 }
 
+/** Returns a pointer to a cropped X Histrogram to be used as the X Histogram
+ *  for each of the spectra in the new cropped workspace
+ *  @param workspace :: The workspace from which to get the cropped X Histogram
+ */
+const Kernel::cow_ptr<Mantid::HistogramData::HistogramX>
+ExtractSpectra::getCroppedXHistogram(const API::MatrixWorkspace &workspace) {
+  const auto hist = workspace.histogram(0);
+  auto begin = m_minXIndex;
+  auto end = m_maxXIndex;
+  if (m_isHistogramData) {
+    end -= 1;
+  }
+
+  auto cropped(hist);
+  cropped.resize(end - begin);
+
+  auto xEnd = hist.xMode() == Histogram::XMode::Points ? end : end + 1;
+  cropped.mutableX().assign(hist.x().begin() + begin, hist.x().begin() + xEnd);
+  return cropped.sharedX();
+}
+
+/** Crops the given workspace in accordance with m_minX and m_maxX
+ *  @param workspace :: The output workspace to crop
+ *  @param XHistogram :: Pointer to the cropped X Histogram to be used in the output workspace
+ *  @param index :: The workspace index of the spectrum
+ */
+void ExtractSpectra::cropCommon(API::MatrixWorkspace &workspace,
+                                Kernel::cow_ptr<Mantid::HistogramData::HistogramX> XHistogram, int index) {
+  const auto hist = workspace.histogram(index);
+  auto begin = m_minXIndex;
+  auto end = m_maxXIndex;
+  if (m_isHistogramData) {
+    end -= 1;
+  }
+
+  auto cropped(hist);
+  cropped.resize(end - begin);
+
+  cropped.setSharedX(XHistogram);
+
+  if (cropped.sharedY())
+    cropped.mutableY().assign(hist.y().begin() + begin, hist.y().begin() + end);
+  if (cropped.sharedE())
+    cropped.mutableE().assign(hist.e().begin() + begin, hist.e().begin() + end);
+  if (cropped.sharedDx())
+    cropped.mutableDx().assign(hist.dx().begin() + begin, hist.dx().begin() + end);
+
+  workspace.setHistogram(index, cropped);
+}
+
+/** Zeroes all data points outside the X values given
+ *  @param workspace :: The output workspace to crop
+ *  @param index :: The workspace index of the spectrum
+ */
+void ExtractSpectra::cropRagged(MatrixWorkspace &workspace, int index) {
+  auto &Y = workspace.mutableY(index);
+  auto &E = workspace.mutableE(index);
+  const size_t size = Y.size();
+  size_t startX = this->getXMinIndex(index);
+  if (startX > size)
+    startX = size;
+  for (size_t i = 0; i < startX; ++i) {
+    Y[i] = 0.0;
+    E[i] = 0.0;
+  }
+  size_t endX = this->getXMaxIndex(index);
+  if (endX > 0)
+    endX -= m_isHistogramData;
```

Comment:
I thought this was a bool. Why are we subtracting it here?

Reply:
Yes, so this is happening in quite a few places in this file (``cropRagged`` was an existing function which I've moved). I've not replicated it in my new code, but I can go through and remove the occurrences in the rest of the file 👍 I agree it's confusing to read, even though you can technically do it (bool is just treated as 0 or 1).